src/core/config/ip6.h

```diff
@@ -177,4 +177,33 @@
 #define OPENTHREAD_CONFIG_UNSECURE_TRAFFIC_MANAGED_BY_STACK_ENABLE 0
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_IP6_MAX_MULTICAST_LISTENER_REGISTRATION_NUM
+ *
+ * The maximum number of total IPv6 multicast addresses (with scope >= 4) that may be externally added by the device
+ * itself and registered by its children. At least should be the same as OPENTHREAD_CONFIG_IP6_MAX_EXT_MCAST_ADDRS for
+ * MTD device. And allocate more for FTD in case it would proxy additional Multicast address its MTD children
+ * registered.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_IP6_MAX_MULTICAST_LISTENER_REGISTRATION_NUM
+#if OPENTHREAD_FTD
+#define OPENTHREAD_CONFIG_IP6_MAX_MULTICAST_LISTENER_REGISTRATION_NUM (OPENTHREAD_CONFIG_IP6_MAX_EXT_MCAST_ADDRS * 3)
```

Comment:
Just to be clear, this value represents the total number of unique IPv6 multicast addresses that are registered by the local device as well as all children? For example, if each child registered a unique multicast address, that would mean one entry per child?

Reply:
Yes and yes.